src/wallet/context.h

```diff
@@ -23,6 +34,9 @@ class Chain;
 struct WalletContext {
     interfaces::Chain* chain{nullptr};
     ArgsManager* args{nullptr};
+    RecursiveMutex wallets_mutex;
```

Comment:
4be544c7ec08a81952fd3f4349151cbb8bdb60e8

It seems this can be a `Mutex`, just a suggestion for follow-up.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19101#discussion_r646946660

> [4be544c](https://github.com/bitcoin/bitcoin/commit/4be544c7ec08a81952fd3f4349151cbb8bdb60e8)
> 
> It seems this can be a `Mutex`, just a suggestion for follow-up.

Thanks went ahead and made the change here. It's not an immediately obvious change because the mutex is held while calling wallet_load_fns callbacks, but looking at the callbacks registered in GUI code it seems like these will never change the list of wallets so it seems good change to get rid of the recursive mutex now to prevent more complexity in the future

Reply:
I also tested `Mutex` and built it with tsan.